VirtIO/VirtIORing.c

```diff
@@ -349,16 +351,14 @@ static bool virtqueue_kick_prepare_split(struct virtqueue *_vq)
     u16 old, new;
     KeMemoryBarrier();
 
-    wrap_around = (vq->num_added_since_kick >= (1 << 16));
-
-    old = (u16)(vq->master_vring_avail.idx - vq->num_added_since_kick);
-    new = vq->master_vring_avail.idx;
-    vq->num_added_since_kick = 0;
-
     if (_vq->vdev->event_suppression_enabled) {
+        wrap_around = (vq->num_added_since_kick >= (1 << 16));
```

Comment:
While I'm at it, is this code right? Isn't 1<<16 ~65k? That seems absurdly large, wouldn't we want to be talking about wrapping around the size of the ring? or am I missing something?

Reply:
We need to look at it. Similar code exists in Linux kernel virtio_ring.c